### PR TITLE
Added example for STM32F411 using SPI

### PR DESCRIPTION
**Pull request by blaz-r**
I've made an example for STM32F411 micro-controller, also know as black pill. It's using smart-leds, stm32f4xx_hal and ws2812 SPI driver.

I've also written a readme with quick-start guide about toolchain and in some parts linked useful info, including a more detailed readme in my repository.

### CODE REVIEW

File: stm32f411-examples/examples/stm32f411_example_spi_rainbow.rs [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        }
+    }
+    loop {
+        continue;
+    }
+}
+
+/// Input a value 0 to 255 to get a color value
+/// The colours are a transition r - g - b - back to r.
+fn wheel(mut wheel_pos: u8) -> RGB8 {
```
Comment (sajattack) [inline stm32f411-examples/examples/stm32f411_example_spi_rainbow.rs]: So this is really on us for not updating the examples, but you can do this in a better way with the hsv conversion function https://github.com/smart-leds-rs/smart-leds/blob/b5cf6f0a8770b11a35a6589218916e0ac05c6f6d/src/hsv.rs#L19

**Comment by sajattack** [comment]
Looks great, thanks! Let me know if you feel like using the HSV thing instead of the wheel function, something we should probably do going forward but not a huge deal so I could go either way.

**Comment by blaz-r** [comment]
Oh yeah, HSV would work nicely here. Well, I mostly wanted to submit an example showing how it works with stm32f4, so I didn't really think of HSV.

I'll try to make it work with HSV, it'll just take a bit since I won't really have time untill next weekend :)

**Review by david-sawatzke — COMMENTED** [review]
This looks good, but if you're already providing probe-run support, it might be better to tell people just to use it instead of having to install openocd & arm-none-eabi-gdb stuff

**Comment by blaz-r** [comment]
Ok, I will also fix that. That probe-run comes with hal crate config file, so I kept that line. To load code I followed embedded rust book, so I was not familiar, until now, with what probe-run actually does. I hope that's not too big of a problem but I'm pretty new to embedded rust, 
Main reason for writing the entire quick-guide is that I didn't really find much recent examples for stm32f411 blackpill and wanted to share some info to maybe help others, so I really appreciate your input :).

So, when I'll have access to hardware and time, I'll implement HSV as said before, then test and add paragraph for probe-run.

**Comment by david-sawatzke** [comment]
Ah alright: 

So for using `probe-run` install it with `cargo install probe-run`. The you can run it with `cargo run --release`, which compiles & flashes it onto the MCU.

**Comment by blaz-r** [comment]
Ok.
I now changed rainbow to use HSV and updated readme to also contain info about probe-run.
Rainbow effect is not exactly the same, but is very similar. 

Let me know if anything still needs to be changed.

**Review by sajattack — APPROVED** [review]

**Review by david-sawatzke — APPROVED** [review]
Thank you, works great!

**Pull request merged**